### PR TITLE
LPS-115411 create numeric field type report processor

### PR DESCRIPTION
**Pull request by marcosapmf**
Related issue: https://issues.liferay.com/browse/LPS-115411

### CODE REVIEW

**Comment by natocesarrego** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/report/NumericDDMFormFieldTypeReportProcessor.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+		double sum = summaryJSONObject.getDouble("sum", 0);
+
+		if (ddmFormInstanceReportEvent.equals(
+				DDMFormInstanceReportConstants.EVENT_ADD_RECORD_VERSION)) {
+
+			if (!summaryJSONObject.has("max") ||
+				(numberValue.doubleValue() > summaryJSONObject.getDouble(
+					"max"))) {
+
+				summaryJSONObject.put("max", numberValue.doubleValue());
```
[the remark this answers is not in the document]
Comment (marcosapmf) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/report/NumericDDMFormFieldTypeReportProcessor.java:79]: @natocesarrego, yes. The idea is to set `max` value if it does not exist yet or the existing `max` is lesser than `numberValue`, thus `numberValue` becomes the new `max`

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/report/NumericDDMFormFieldTypeReportProcessor.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+				"max", maxValue.doubleValue()
+			).put(
+				"min", minValue.doubleValue()
+			);
+
+			sum -= numberValue.doubleValue();
+		}
+
+		summaryJSONObject.put(
+			"average", sum / jsonObject.getInt("totalEntries")
```
[the remark this answers is not in the document]
Comment (marcosapmf) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/report/NumericDDMFormFieldTypeReportProcessor.java]: @natocesarrego, I've fixed that in the new force push :)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/report/NumericDDMFormFieldTypeReportProcessor.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+		if (Validator.isNull(value)) {
+			return null;
+		}
+
+		return GetterUtil.getNumber(value);
+	}
+
+	private Stream<Number> _getNumberValues(
+		DDMFormInstance ddmFormInstance, long formInstanceRecordId,
+		String fieldName) {
```
[the remark this answers is not in the document]
Comment (marcosapmf) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/report/NumericDDMFormFieldTypeReportProcessor.java]: @natocesarrego, I've fixed that in the new force push :)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/report/NumericDDMFormFieldTypeReportProcessor.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (natocesarrego) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/report/NumericDDMFormFieldTypeReportProcessor.java]: Great job! Thanks. 😄

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/report/NumericDDMFormFieldTypeReportProcessor.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (natocesarrego) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/report/NumericDDMFormFieldTypeReportProcessor.java]: Thanks! 😄

**Comment by natocesarrego** [comment]
ci:forward

**Pull request closed without merging**